Customizable GAGS Witch Hat

Kepteyn opened the pull request:
## About The Pull Request

Implements new sprites for the witch (Marisa) hat, divided into three greyscale icon states - base, bow, and charms. The hat can have its colours customized at an AutoDrobe. The not fake version has also been changed in the same way. The default colours are approximately the same to the old hat's.

![A coven of witches showing off their newly colourable hats](https://user-images.githubusercontent.com/94123167/210642213-42e08912-1c11-469e-afda-66ce567280dd.png)
_A coven of witches showing off their newly colourable hats_


## Why It's Good For The Game

More customization options for clothes allow for more variety and greater player creativity.

## Changelog

:cl:
add: added greyscale json configs for witch hat and witch hat worn.
imageadd: Added new, greyscale states to head/wizard.dmi
imagedel: Removed Marisa hat states from head/wizard.dmi
code: changed Marisa hat code in wiz_robes.dm to make them use greyscale.
config: changed greyscale configs to include the witch hat.
/:cl:

Kepteyn committed:
Changes witch hat to greyscale

Kepteyn committed:
Update wizard.dmi

Twaticus commented:
the whole face is covered in the new front facing sprite pls fix, otherwise cool addition

Kepteyn committed:
Reworks front-view sprite

Kepteyn commented:
Changed the front sprite to not completely cover the face:
![witch](https://user-images.githubusercontent.com/94123167/211172015-f9e49a0e-49b5-4c40-b2e2-1d063d0b68dc.png)

Kepteyn committed:
Merge branch 'master' into Witchhat

optimumtact commented:
Conflicted

JohnFulpWillard commented on code/modules/clothing/suits/wiz_robe.dm:
```diff
@@ -149,8 +154,13 @@
 	name = "witch hat"
-	desc = "Strange-looking hat-wear, makes you want to cast fireballs."
-	icon_state = "marisa"
+	desc = "Strange-looking hat-wear. Makes you want to cast fireballs."
+	icon_state = "witch_hat"
+	greyscale_colors = "#343640#e0cab8#e0cab8"
+	greyscale_config = /datum/greyscale_config/witch_hat
+	greyscale_config_worn = /datum/greyscale_config/witch_hat_worn
+	flags_1 = IS_PLAYER_COLORABLE_1
+	dog_fashion = null
```
None of this has to be here because it's a subtype of ``/obj/item/clothing/head/wizard/marisa`` so copies it from that

Kepteyn replied:
Removed the surplus code and tested to ensure it works still. Thank you!

Kepteyn committed:
Fixes conflict and removes excess code

Kepteyn committed:
fixes merge conflict with wizhood

Kepteyn committed:
Merge remote-tracking branch 'upstream/master' into Witchhat

Kepteyn commented:
Sorry for the delay; I've fixed the merge conflict (or at least I'm _pretty sure_ I have? Fairly new to github still). The wizhood and the witch hat are both present now. I also removed some unneeded code and tested to ensure it works still.

san7890 commented:
"Config Update" typically refers to changes to server configuration (i.e. anything in the /config folder) that server operators can use to alter the game to their whim and desires- greyscale configs aren't really covered by that. I know there's some confusion with the shared terminology, no worries.
